src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4520,34 +4519,23 @@ private ParameterSyntax ParseParameter()
                 //
                 // We can potentially merge that with an immediately following !! or an immediately following !=
                 // All other cases are in error.
-                var firstExclamation = this.EatToken();
+                identifier = AddTrailingSkippedSyntax(identifier, this.AddError(this.EatToken(), ErrorCode.ERR_ParameterNullCheckingNotSupported));
                 if (this.CurrentToken.Kind is SyntaxKind.ExclamationToken)
                 {
                     // have two !'s in a row.  Merge them (reporting any errors if they cannot merge properly).
-                    exclamationExclamationToken = MergeAdjacent(firstExclamation, this.EatToken(), SyntaxKind.ExclamationExclamationToken);
+                    identifier = AddTrailingSkippedSyntax(identifier, this.EatToken());
```

Comment:
nit: the comment above seems out of date (we're no longer merging exclamation tokens)

Reply:
Updated the comments.